cinder/scheduler/filter_scheduler.py

```diff
@@ -109,6 +109,36 @@ def schedule_create_volume(self, context, request_spec, filter_properties):
                                          filter_properties,
                                          allow_reschedule=True)
 
+    def find_backend_for_connector(self, context, connector, request_spec):
+        weighed_backends = self._get_weighted_candidates(context, request_spec)
+
+        if not weighed_backends:
+            raise exception.NoValidBackend(reason=_("No weighed backends "
+                                                    "available"))
+
+        def _backend_matches_connector(bck, conn):
+            key = 'connection_capabilities'
+            backend_capabilities = set(bck.obj.capabilities.get(key, []))
+            connector_capabilities = set(conn.get(key, []))
```

Comment:
In `initialize_connection()`, we don't use a default but instead don't check if the `connector` doesn't have a `connection_capabilities` attribute. We should make sure it behaves the same. Either say "no capabilities set means it's old code and we should not check" or "no capabilities -> none requested -> we can't check -> deny".